pkg/services/ssosettings/validation/oauth_validators.go

```diff
@@ -17,6 +17,20 @@ func AllowAssignGrafanaAdminValidator(info *social.OAuthInfo, requester identity
 	return nil
 }
 
+func OrgMappingValidator(info *social.OAuthInfo, requester identity.Requester) error {
+	if len(info.OrgMapping) > 0 && !requester.GetIsGrafanaAdmin() {
+		return ssosettings.ErrInvalidOAuthConfig("Organization mapping can only be updated by Grafana Server Admins.")
+	}
+	return nil
+}
+
+func OrgAttributePathValidator(info *social.OAuthInfo, requester identity.Requester) error {
+	if info.OrgAttributePath != "" && !requester.GetIsGrafanaAdmin() {
```

Comment:
Similar to my previous comment, I think we should better check if `OrgAttributePath` has been modified by non Grafana admins.